### PR TITLE
XN-691: Added functionality to update status of meeting registrants

### PR DESCRIPTION
**Pull request by ofalei**
- extended the Zoom client with a new method to update the status of a meeting registrant

### CODE REVIEW

File: meeting_update_registrant.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+	// MeetingRegistrantDetails contains identification of the meeting registrant
+	MeetingRegistrantDetails struct {
+		ID    string `json:"id"`
+		Email string `json:"email"`
+	}
+
+	// MeetingRegistrantsStatus contains options for updating status
+	MeetingRegistrantsStatus struct {
+		MeetingID     int                               `json:"-" url:"-"`
```
Comment (karhidle) [inline meeting_update_registrant.go:31]: why do we need the url?

**Review by cliffom — APPROVED** [review]

**Review by karhidle — APPROVED** [review]

File: meeting_update_registrant.go (hunk 1 above)
Comment (ofalei) [reply meeting_update_registrant.go:31]: Zoom API accepts parameters through Query string and through request body. This library uses [go-querystring](https://github.com/google/go-querystring) package which rely on `url` tag to identify query parameters.

File: meeting_update_registrant.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	// MeetingRegistrantStatusActionCancel defines the action to cancel the registrant of the meeting
+	MeetingRegistrantStatusActionCancel MeetingRegistrantStatusActionType = "cancel"
+
+	// MeetingRegistrantStatusActionDeny defines the action to deny the registrant of the meeting
+	MeetingRegistrantStatusActionDeny MeetingRegistrantStatusActionType = "deny"
+)
+
+type (
+	// MeetingRegistrantStatusActionType defines the type of actions on the registrant status
+	MeetingRegistrantStatusActionType string
```
Comment (geegonzalez) [inline meeting_update_registrant.go:21]: Lol, I was going to suggest making an enum of the different action types, but then I realized that Go doesn't have enums.

File: meeting_update_registrant.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	// MeetingRegistrantDetails contains identification of the meeting registrant
+	MeetingRegistrantDetails struct {
+		ID    string `json:"id"`
+		Email string `json:"email"`
+	}
+
+	// MeetingRegistrantsStatus contains options for updating status
```
Comment (geegonzalez) [inline meeting_update_registrant.go:29]: The comments are helpful.

File: meeting_update_registrant.go (hunk 2 above)
Comment (mpurdon) [reply meeting_update_registrant.go:21]: Yeah, you can't be trusted with the wild powers of the ENUM! We should prolly add an IsValid() method though. Something like:
```
func (at MeetingRegistrantStatusActionType) IsValid() error {
    switch at {
    case MeetingRegistrantStatusActionApprove,
         MeetingRegistrantStatusActionCancel,
         MeetingRegistrantStatusActionDeny:
        return nil
    }

    return errors.New("invalid action type")
}
```

**Pull request merged**